pdx213: add panel and flash

erebion opened the pull request:
These commits make the panel work, but I'm not sure what I'll have to change to make this acceptable :thinking:

Also, even though I've spent a lot of time trying to make make everything correct, I'm pretty sure it isn't all correct...

so... idk

Commit:
arm64: dts: qcom: sm6350-lena: Add display nodes and enable GPU

Add required nodes to enable panel and GPU on Xperia 10 III.

Signed-off-by: Konrad Dybcio <konrad.dybcio@somainline.org>
Co-developed-by: erebion <erebion@erebion.eu>
Signed-off-by: erebion <erebion@erebion.eu>

MarijnS95 committed:
drm/panel/samsung-sofef01: Add panel driver for Sony Xperia 5 / 10 II

This SOFEF01-M Display-IC driver supports two modes with different
compatibles to differentiate between slightly different physical sizes
(panels) found on the Xperia 5 (6.1") and 10 II (6.0").

It is currently also used to hardcode significantly higher fake porches
for the Xperia 5, which are unused in transfers due to this being a
command-mode panel but do have an effect on the clock rates set by
dsi_host.c.  Without higher clock rates this panel fails to achieve
60fps and has significant tearing artifacts, while the same calculated
clock rate works perfectly fine on the Xperia 10 II.

Signed-off-by: Marijn Suijten <marijn.suijten@somainline.org>

Commit:
drm/panel: panel-samsung-sofef01: transition to mipi_dsi wrapped functions

Changes the samsung-sofef00 panel to use multi style functions for
improved error handling.

(basically trying to do the same as https://github.com/torvalds/linux/commit/d6a4da523deba1814a45a593c271c32efedc39f3 but for this specific panel)

Signed-off-by: erebion <erebion@erebion.eu>

Commit:
defconfig: add config that enables the panel on Xperia 10 III

Commit:
arm64: dts: qcom: sm6350: Add refgen regulator

Add a node for the refgen regulator found on SM6350 which is used as a
supply for the MDSS.

Signed-off-by: erebion <erebion@erebion.eu>

z3ntu commented on arch/arm64/boot/dts/qcom/sm6350-sony-xperia-lena-pdx213.dts:
```diff
@@ -343,6 +350,77 @@
+	status = "okay";
+	zap-shader {
+		firmware-name = "qcom/sm6350/sony/pdx213/a615_zap.mbn";
+	};
```
Move this zap-shader into its own node outside of &gpu
```
&gpu_zap_shader {
    firmware-name = "qcom/sm6350/sony/pdx213/a615_zap.mbn";
};
````

erebion replied:
huh

Would you mind elaborating? I did that at first, then saw it somewhere like this. xD

z3ntu replied:
if you saw it somewhere, it's older style. Just change it ;)

erebion replied:
done (amended commit), thanks

z3ntu commented on arch/arm64/boot/dts/qcom/sm6350-sony-xperia-lena-pdx213.dts:
```diff
@@ -343,6 +350,77 @@
+	panel_reset_pin: panel-reset-state {
+		pins = "gpio9";
+		function = "func1";
+		qcom,drive-strength = <0x02>;
```
don't think this property exists on mainline?

erebion replied:
Whoops. Copy/Paste error, wanted to remove that. -.-

thanks

erebion replied:
done (amended commit), thanks

z3ntu commented on arch/arm64/boot/dts/qcom/sm6350-sony-xperia-lena-pdx213.dts:
```diff
@@ -28,7 +29,6 @@
 			height = <2520>;
 			stride = <(1080 * 4)>;
 			format = "a8r8g8b8";
-			clocks = <&gcc GCC_DISP_AXI_CLK>;
```
is this change necessary for dsi panel to work?

erebion replied:
I'm not sure, that's a change by Konrad Dybcio, I cherry-picked that commit when you said it might be helpful to start with that one.

Haven't changed the change as I'm not sure about it's meaning-

z3ntu replied:
then try and remove this removal, I think it should still work

erebion replied:
changed it, but haven't yet tried it, will do that now

erebion replied:
Without that change (with that line added), the phone will only show the Sony logo booting up. Once I remove the line, I see 8x Tux happily looking out from the screen. :)

z3ntu replied:
huh, that's weird... That should really only keep the clock alive and then sort of handover to mdss which properly drives the display.

erebion replied:
Idk, but leaving it in breaks the panel :shrug:

z3ntu replied:
ok weird, then keep it removed I guess.

z3ntu commented on arch/arm64/boot/dts/qcom/sm6350-sony-xperia-lena-pdx213.dts:
```diff
@@ -44,6 +44,13 @@
+		compatible = "regulator-fixed";
+		regulator-name = "lcdb_dummy";
+		regulator-min-microvolt = <5500000>;
+	regulator-max-microvolt = <5500000>;
```
fix the indentation, add regulator-boot-on; regulator-always-on;

erebion replied:
done (amended commit), thanks

z3ntu commented on arch/arm64/boot/dts/qcom/sm6350-sony-xperia-lena-pdx213.dts:
```diff
@@ -343,6 +350,77 @@
+		vsp-supply = <&lcdb_dummy>;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&te_gpio &panel_reset_pin>;
```
first pinctrl-0, then pinctrl-names

erebion replied:
done (amended commit), thanks

z3ntu replied:
I didn't check the inbetween commits, but the full diff is still the wrong way around

erebion replied:
done

z3ntu commented on arch/arm64/boot/dts/qcom/sm6350-sony-xperia-lena-pdx213.dts:
```diff
@@ -343,6 +350,77 @@
+
+&mdss_dsi0 {
+	vdda-supply = <&pm6350_l22>;
+	status = "okay";
```
newline before status

erebion replied:
done (amended commit), thanks

z3ntu commented on arch/arm64/boot/dts/qcom/sm6350-sony-xperia-lena-pdx213.dts:
```diff
@@ -343,6 +350,77 @@
+&mdss_dsi0_out {
+	remote-endpoint = <&panel_in>;
+	data-lanes = <0 1 2 3>;
+	refgen-supply = <&refgen>;
```
this should be in sm6350.dtsi, pretty sure

erebion replied:
The whole &mdss_dsi0_out block you mean?

z3ntu replied:
refgen-supply only

z3ntu replied:
actually this refgen-supply should be a property of `mdss_dsi0` and not `mdss_dsi0_out`. 

Were you saying in https://github.com/sm6350-mainline/linux/pull/1#issuecomment-3250842228 that when you initially opened the PR it was working fine, but not anymore with the changes as-is now?

erebion replied:
No, just generally observed that from dmesg and wondered whether it is relevant to this MR.

erebion replied:
done

z3ntu replied:
was still in a random place, fixed it up now

z3ntu commented on arch/arm64/boot/dts/qcom/sm6350-sony-xperia-lena-pdx213.dts:
```diff
@@ -343,6 +350,77 @@
+
+&mdss_dsi0_phy {
+	vdds-supply = <&pm6350_l11>;
+	status = "okay";
```
newline before status

erebion replied:
done (amended commit), thanks

z3ntu commented on arch/arm64/boot/dts/qcom/sm6350.dtsi:
```diff
@@ -822,6 +822,11 @@
+                refgen: regulator@88e7000 {
+                        compatible = "qcom,sm6350-refgen-regulator";
+                        reg = <0x0 0x088e7000 0x0 0x60>;
+                };
```
this node needs to go way further down, it should be sorted by that second value in reg, so 0x088e7000, the ones around have one less zero

erebion replied:
changed it, hope it's correct now :)

z3ntu replied:
still not, but I can fix it up myself

z3ntu commented:
Please keep the nodes in the dts ordered, e.g. mdss and gpu are a bit in random places now, they should be ordered alphabetically

erebion commented:
Error:

```
aux_bridge.aux_bridge aux_bridge.aux_bridge.0: error -ENODEV: failed to acquire drm_bridge
```

Warnings:

```
platform ae94000.dsi: deferred probe pending: msm_dsi: Failed to get supply 'refgen'
qcom-rpmhpd 18200000.rsc:power-controller: sync_state() pending due to ae94000.dsi
```

Related to these changes?

erebion commented:
> Please keep the nodes in the dts ordered, e.g. mdss and gpu are a bit in random places now, they should be ordered alphabetically

I've sorted things now

erebion commented:
Is there a standard for the firmware file path? I just guessed what seemed fine, as I couldn't find anything.

z3ntu commented:
> Is there a standard for the firmware file path? I just guessed what seemed fine, as I couldn't find anything.

Uhm, it's a bit free-form. Usually something like `qcom/$soc/$vendor/$model/foo.mbn` but regarding capitalization etc, dunno. It's definitely not very consistent upstream but changing it also means breaking user space, so it's usually not touched.

z3ntu commented on arch/arm64/boot/dts/qcom/sm6350-sony-xperia-lena-pdx213.dts:
```diff
@@ -277,6 +286,10 @@
 
+&gmu {
+	status = "okay";
+};
```
No need for this, gmu is enabled by default (i.e. no `status = "disabled"` in `sm6350.dtsi`)

erebion replied:
done

z3ntu commented:
I've cleaned up the commits somewhat and bring them into style how they're expected to look. Please take a look so you can see what kind of format we need.

It has also changed some things to please re-test to see if it still works.

Then additionally we need Signed-off-by for the commits, I've marked them FIXME in the commit message https://www.kernel.org/doc/html/latest/process/submitting-patches.html

The doc says "using a known identity (sorry, no anonymous contributions.)", I think just "erebion" should be fine if you don't want to use your (or **a**) name. You're not `anonymous123 <throwaway@example.org>` afterall.

erebion commented:
> I've cleaned up the commits somewhat and bring them into style how they're expected to look. Please take a look so you can see what kind of format we need.

I'm not entirely sure. Some parts are alphabetically sorted, but then there are other parts above those that are not alphabetically sorted. Still not sure what the pattern is.

>
> It has also changed some things to please re-test to see if it still works.

Compiling now...


> 
> Then additionally we need Signed-off-by for the commits, I've marked them FIXME in the commit message https://www.kernel.org/doc/html/latest/process/submitting-patches.html
> 
> The doc says "using a known identity (sorry, no anonymous contributions.)", I think just "erebion" should be fine if you don't want to use your (or **a**) name. You're not `anonymous123 <throwaway@example.org>` afterall.

Done. Thanks.

erebion commented:
> Compiling now...

Works just as before, including the same error and two warnings as before (https://github.com/sm6350-mainline/linux/pull/1#issuecomment-3250842228)

Commit:
arm64: dts: qcom: sm6350-lena: Add flash LED

Add a node for the camera flash LED found on the Xperia 10 III

Signed-off-by: erebion <erebion@erebion.eu>

z3ntu commented:
FYI refgen for sm6350 was posted upstream by Dmitry: https://lore.kernel.org/linux-arm-msm/20250921-refgen-v1-5-9d93e64133ea@oss.qualcomm.com/

Commit:
wip: defconfig

z3ntu commented:
What was the latest state now, it was working right? Can we merge (the majority of) the patches?

erebion commented:
> What was the latest state now, it was working right? Can we merge (the majority of) the patches?

Panel works, but the colours look a little bit washed out. If there's a way to increase contrast, I should probably do so. Not sure whether that's because of these patches or because of something else.

z3ntu commented:
Thanks, I've pushed the commits (apart from your defconfig changes) to `sm6350-6.17.y`!

https://github.com/sm6350-mainline/linux/commits/sm6350-6.17.y/

Feel free to close the PR.

erebion commented:
> Thanks, I've pushed the commits (apart from your defconfig changes) to `sm6350-6.17.y`!
> 
> https://github.com/sm6350-mainline/linux/commits/sm6350-6.17.y/
> 
> Feel free to close the PR.

Cool, thanks! :)
